R/methods-Chromatograms.R

```diff
@@ -80,10 +89,18 @@ setMethod("[", "Chromatograms",
               ## Return a single element as a Chromatogram
               if (length(i) == 1 & length(j) == 1)
                   return(x@.Data[i, j, drop = TRUE][[1]])
+              pd <- x@phenoData
               ## Multiple elements, return type depends on drop.
               x <- x@.Data[i = i, j = j, drop = drop]
-              if (!drop)
+              if (!drop) {
                   x <- as(x, "Chromatograms")
+                  pd <- pd[j, ]
+                  ## Drop levels
+                  pData(pd) <- droplevels(pData(pd))
+                  ## set row names
+                  rownames(pd) <- colnames(x)
```

Comment:
It is a bit confusing for me. Sometimes the `phenoData` set the `colnames` of `.Data` and here it is the other way around. Which one is more important? Is it really necessary to overwrite this information again and again?